miio/fan_miot.py

```diff
@@ -0,0 +1,285 @@
+import enum
+from typing import Any, Dict
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .fan_common import FanException, MoveDirection, OperationMode
+from .miot_device import MiotDevice
+
+MODEL_FAN_P9 = "dmaker.fan.p9"
+MODEL_FAN_P10 = "dmaker.fan.p10"
+
+MIOT_MAPPING = {
+    MODEL_FAN_P9: {
+        # Source https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:fan:0000A005:dmaker-p9:1
+        "power": {"siid": 2, "piid": 1},
+        "fan_level": {"siid": 2, "piid": 2},
+        "child_lock": {"siid": 3, "piid": 1},
+        "fan_speed": {"siid": 2, "piid": 11},
+        "swing_mode": {"siid": 2, "piid": 5},
+        "swing_mode_angle": {"siid": 2, "piid": 6},
+        "power_off_time": {"siid": 2, "piid": 8},
+        "buzzer": {"siid": 2, "piid": 7},
+        "light": {"siid": 2, "piid": 9},
+        "mode": {"siid": 2, "piid": 4},
+        "set_move": {"siid": 2, "piid": 10},
+    },
+    MODEL_FAN_P10: {
+        # Source https://miot-spec.org/miot-spec-v2/instance?type=urn:miot-spec-v2:device:fan:0000A005:dmaker-p10:1
+        "power": {"siid": 2, "piid": 1},
+        "fan_level": {"siid": 2, "piid": 2},
+        "child_lock": {"siid": 3, "piid": 1},
+        "fan_speed": {"siid": 2, "piid": 10},
+        "swing_mode": {"siid": 2, "piid": 4},
+        "swing_mode_angle": {"siid": 2, "piid": 5},
+        "power_off_time": {"siid": 2, "piid": 6},
+        "buzzer": {"siid": 2, "piid": 8},
+        "light": {"siid": 2, "piid": 7},
+        "mode": {"siid": 2, "piid": 3},
+        "set_move": {"siid": 2, "piid": 9},
+    },
+}
+
+
+class OperationModeMiot(enum.Enum):
+    Normal = 0
+    Nature = 1
+
+
+class FanStatusMiot:
+    """Container for status reports from the Xiaomi Mi Smart Pedestal Fan DMaker P9/P10."""
```

Comment:
Could you add a pretty-formatted response for the status command (use `--debug` to get the output), so that anyone without access to a device can reason about the property values.